packages/blocks/src/api/registration.js

```diff
@@ -160,6 +160,23 @@ export const serverSideBlockDefinitions = {};
 // eslint-disable-next-line camelcase
 export function unstable__bootstrapServerSideBlockDefinitions( definitions ) {
 	for ( const blockName of Object.keys( definitions ) ) {
+		// Don't overwrite if already set. It covers the case when metadata
+		// was initialized from the server.
+		if ( serverSideBlockDefinitions[ blockName ] ) {
+			// We still need to polyfill `apiVersion` for WordPress version
+			// lower than 5.7. If it isn't present in the definition shared
+			// from the server, we try to fallback to the definition passed.
+			// @see https://github.com/WordPress/gutenberg/pull/29279
+			if (
+				serverSideBlockDefinitions[ blockName ].apiVersion ===
+					undefined &&
+				definitions[ blockName ].apiVersion
```

Comment:
I think I spotted a few more differences on the server definition:
- Missing `editorStyle`
- Sometimes items in `supports`, get expanded in block `attributes` (I think this is okay)
- Missing `style`

Reply:
It might be worth setting up a test case with all block registration items set and then compare to see what's returned.

Reply:
> Missing `editorStyle`
> Missing `style`

At the moment it's used only on the server to enqueue CSS files, example:
https://github.com/WordPress/gutenberg/blob/master/docs/designers-developers/developers/block-api/block-metadata.md#editor-style

In the future, we might need it on the client to lazy load CSS or JS files for the block.

> Sometimes items in `supports`, get expanded in block `attributes` (I think this is okay)

Yes, it's used quite extensively for features built-in in the block editor like alignment, custom class name, anchor, fonts, colors, and other stuff that Global Styles offer. Most (hopefully all) of the hooks that add those fields for `attributes` respect the version created by developers.

Reply:
Great, sounds like this should be relatively safe to try again. ✨